src/main/scala/beam/agentsim/agents/ridehail/ParallelAlonsoMoraAlgForRideHail.scala

```diff
@@ -0,0 +1,135 @@
+package beam.agentsim.agents.ridehail
+
+import beam.agentsim.agents.ridehail.AlonsoMoraPoolingAlgForRideHail._
+import beam.router.BeamSkimmer
+import org.jgrapht.graph.DefaultEdge
+
+import scala.collection.immutable.List
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class ParallelAlonsoMoraAlgForRideHail(
+  demand: List[CustomerRequest],
+  supply: List[VehicleAndSchedule],
+  timeWindow: Map[MobilityServiceRequestType, Int],
+  radius: Int,
+  implicit val skimmer: BeamSkimmer
+) {
+
+  private def parallelBuildOfRTVGraph(): AlonsoMoraPoolingAlgForRideHail.RTVGraph = {
+    import scala.concurrent.ExecutionContext.Implicits.global
+    import scala.collection.mutable.{ListBuffer => MListBuffer}
+    val rTvG = AlonsoMoraPoolingAlgForRideHail.RTVGraph(classOf[DefaultEdge])
+    Future.sequence(supply.withFilter(_.getFreeSeats > 0).map { v =>
+      Future {
+        val vertices = MListBuffer.empty[RTVGraphNode]
```

Comment:
Is it possible to extract this to a function (the body of Future) to be able to test independently and improve readability?